### ISSUE TITLE
B020 Bug

### ISSUE THREAD

**Issue by btjones0** [PyCQA/flake8-bugbear#262]
This is similar to #235 but with sort keys.

```python
for student in sorted(students, key=lambda student: student.last_name):
    ...
```

results in B020 but it seems like it should be fine.

**Comment by cooperlees** [comment]
Yeah, I agree if we can find a way to see the scope of student in the lambda I'd happy take that PR.

On the contrary, I do argue tho, it would do nothing but help readability to not use the name student there in the lambda.

```python
for student in sorted(students, key=lambda student_to_sort: student_to_sort.last_name):
```

**Comment by FozzieHi** [comment]
@cooperlees Seems to be fixed now so can be closed :+1:

**Comment by cooperlees** [comment]
Thanks! Do we know the PR or how did you test it to see it pass?

**Comment by FozzieHi** [comment]
@cooperlees Just copied the example above and ran flake8 on it.

Looks like this issue is actually a duplicate of #256 which was fixed in #259 although was not actually released when this issue was made. With `git checkout b1e4ef296ec6afb6df788d81db0953cf5fd79566~1` the issue is there, and with `git checkout b1e4ef296ec6afb6df788d81db0953cf5fd79566` the issue is fixed.

**Comment by cooperlees** [comment]
Many thanks explaining in case someone disagrees in the future with us for closing this!
